app/src/main/res/values/styles.xml

```diff
@@ -8,6 +8,7 @@
         <!-- Android system styling -->
         <item name="searchViewStyle">@style/SearchViewStyle</item>
         <item name="autoCompleteTextViewStyle">@style/AutoCompleteTextViewStyle</item>
+        <item name="android:textAlignment">viewStart</item>
```

Comment:
Any change in `styles.xml` seems to also affect A-C components, as pointed out in [this](https://bugzilla.mozilla.org/show_bug.cgi?id=1811530) ticket.

![Screenshot_20230202_151103](https://user-images.githubusercontent.com/35462038/216334066-8753cc76-94b8-4d34-8a36-7b5600242a21.png)

Should we file a ticket to fix the A-C layouts using `0dp` width first?

Reply:
Good catch! I'm wondering if we should consider not making the `textAlignment` a global attribute, but instead making it specific to Fenix's text styles (e.g. `NeutralButton`, `DialogStyleBase`).

Have you explored that direction already @mavduevskiy ?

Reply:
Good catch indeed! Thanks, @Alexandru2909, we should do something with the A-C side first.

The problem with A-C isn't the `0dp`. The buttons have the default text gravity – center, and again `textAlignment` overrides it. Interestingly enough, the `NeutralButton` mentioned by @jonalmeida has the center `textAlignment`, so buttons in fenix didn't break. I would hope that A-C buttons would have some general button style, but I don't see it. 

Jonathan, I understand your thinking, but is it possible? We want to make sure that all the text views in the app follow the expected direction. If we make it specific to particular fenix style, it doesn't make all the elements inside fenix follow that style and does not affect any AC texts.

Also, taking math into consideration: the rule for the majority of the text elements is to align with the view start, and center position – is for minority. So, setting a `viewStart` as a general rule and adding `center` in places where it's needed seems like the lesser code approach.  A-C atm has 8 `textviews` with manually set alignment, as 7 of them are `viewStart`.

A-C doesn't have a universal style for buttons and has a few gravity center textViews. It looks to me I should just change the text alignment for elements that should position text in center, so they would support RTL languages with a global parameters.

Reply:
You make a fair point. 

Identifying all text would be difficult since that includes buttons and dialogs as well. Not to mention something might seem to be correctly aligned AC, Focus, Sample and Reference Browser, but only when tested in Fenix would you see that alignment needs to be specified in a button that would have otherwise had a text alignment default of center.

Would this flow mean that all AC code current and future should specify the text alignment default for Fenix?

Reply:
Talked with @jonalmeida, a run down:

Problem: introduce changes to AC that would make it resilient to general `textAlignment` style in the app that uses it, while also safeguarding that the next AC components would also follow the protocol of not breaking in the rest of the projects.

Solution: two steps. Patch current AC buttons and elements that position text with `center` gravity to specify text alignment as well, then do a follow up to create a style instead of assigning alignment to each element. 

Bonus: consider making the expected RTL support behavior of AC visible in the AC itself (applying general `textAlignment` there as well)